Implement commands for listing per-cluster CT/SNAT maps

YutaroHayakawa opened the pull request:
Please see individual commits for more details.

```release-note
Implement commands for listing per-cluster CT/SNAT maps
```

YutaroHayakawa committed:
Introduce GetClusterCTMaps method to PerClusterCTMapper

Introduce a new method GetClusterCTMaps which gets all CT maps for
specific cluster.

Signed-off-by: Yutaro Hayakawa <yutaro.hayakawa@isovalent.com>

YutaroHayakawa committed:
Introduce a command to list per-cluster NAT maps

Introduce a new command `cilium bpf nat list cluster <cluster id>` which
dumps per-cluster NAT maps.

Signed-off-by: Yutaro Hayakawa <yutaro.hayakawa@isovalent.com>

YutaroHayakawa committed:
cmd: Cleanup bpf ct list command fixture

In the current `cilium bpf ct list`, we specify the instance of the CT
maps to dump by reserved word "global" or numeric id of the endpoints.
This means we recognize the "type" of the CT maps by checking if the
first argument is string or int and now we're trying to introduce
another type for per-cluster maps.

It is technically possible to introduce compatible command fixture like
follows, but the parsing is unnecessarily complicated.

Global   : cilium bpf ct list global
Enpoints : cilium bpf ct list 12345
Cluster  : cilium bpf ct list cluster 1

Instead, we can explicitly specify the type by the first argument.

Global   : cilium bpf ct list global
Enpoints : cilium bpf ct list endpoint 12345
Cluster  : cilium bpf ct list cluster 1

This breaks the existing command for per-endpoint CT maps, but we
consider this as an acceptable change since it doesn't change output of
the command and cilium command fixture is considered as an cilium
internal things and not a user interface we maintain.

Signed-off-by: Yutaro Hayakawa <yutaro.hayakawa@isovalent.com>

YutaroHayakawa committed:
Introduce a command to list per-cluster CT maps

Introduce a new command `cilium bpf ct list cluster <cluster id>` which
dumps per-cluster CT maps.

Signed-off-by: Yutaro Hayakawa <yutaro.hayakawa@isovalent.com>

YutaroHayakawa committed:
clustermesh: Add a constant of common per-cluster CT map prefix

Signed-off-by: Yutaro Hayakawa <yutaro.hayakawa@isovalent.com>

YutaroHayakawa committed:
Fix a bug of dumpAndRead of ct list command

dumpAndRead takes dump function and argument (args... interface{}) and
calls dump function with args without expand the args slice. Fix it to
expand slice.

Signed-off-by: Yutaro Hayakawa <yutaro.hayakawa@isovalent.com>

nathanjsweet commented on cilium/cmd/bpf_ct_list.go:
```diff
@@ -23,18 +23,23 @@ import (
 var (
 	bpfCtListCmd = &cobra.Command{
-		Use:     "list ( <endpoint identifier> | global )",
+		Use:     "list ( global | endpoint ) [identifier]",
```
Why isn't "cluster" documented here?

YutaroHayakawa replied:
Oops, let me fix this.

nathanjsweet commented on cilium/cmd/bpf_ct_list.go:
```diff
@@ -51,12 +56,46 @@ func init() {
-func getMaps(eID string) []*ctmap.Map {
-	if eID == "global" {
+func parseArgs(args []string) (string, uint32, error) {
+	t := args[0]
```
We should check for at least two args to avoid index errors.

YutaroHayakawa replied:
Good point. I'll fix.

nathanjsweet reviewed (changes requested):
nit and validation concern.

YutaroHayakawa committed:
Implement from_tunnel field handling in flagString

Handle from_tunnel in flagString. So that we can show from_tunnel value
to CLI output.

Signed-off-by: Yutaro Hayakawa <yutaro.hayakawa@isovalent.com>
